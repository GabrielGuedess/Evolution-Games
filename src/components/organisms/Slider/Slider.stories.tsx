import { Story, Meta } from '@storybook/react/types-6-0';

import sliderMock from './mock';
import { Slider, SliderProps } from './Slider';

export default {
  title: 'Organisms/Slider',
  component: Slider,
  args: {
    slides: sliderMock,
  },
} as Meta<SliderProps>;

export const Default: Story<SliderProps> = args => <Slider {...args} />;
