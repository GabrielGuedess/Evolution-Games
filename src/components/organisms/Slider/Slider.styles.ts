import styled, { css } from 'styled-components';

export const Wrapper = styled.section`
  ${({ theme }) => css`
    position: relative;
    width: 100%;
    height: 100vh;

    .swiper {
      width: 100%;
      height: 100%;
    }

    .swiper-slide {
      background: ${theme.colors.mainBg};
    }

    .swiper-pagination {
      margin-bottom: 3rem;

      span.swiper-pagination-bullet.swiper-pagination-bullet-active {
        background-color: ${theme.colors.primary};
        opacity: 1;
      }

      .swiper-pagination-bullet {
        background-color: ${theme.colors.white};
        opacity: 1;
      }
    }
  `}
`;
