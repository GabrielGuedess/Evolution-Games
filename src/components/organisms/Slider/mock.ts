import { SlideProps } from 'components/molecules/Slide/Slide';

export const SliderMock: SlideProps[] = [
  {
    srcSlideImage: '/img/games/theLastOfUs.png',
    altSlideImage: 'The Last Of Us',
    srcHighlightImage: '/img/games/highlightName/nameTheLastOfUs.png',
    altHighlightImage: 'The Last Of Us Highlight',
  },
  {
    srcSlideImage: '/img/games/god.jpg',
    altSlideImage: 'God Of War',
    srcHighlightImage: '/img/games/highlightName/nameGodOfWar.png',
    altHighlightImage: 'God Of War Highlight',
  },
  {
    srcSlideImage: '/img/games/r8.jpg',
    altSlideImage: 'Resident Evil Village',
    srcHighlightImage: '/img/games/highlightName/nameR8.png',
    altHighlightImage: 'Resident Evil Village Highlight',
  },
  {
    srcSlideImage: '/img/games/starWars.jpg',
    altSlideImage: 'Star Wars',
    srcHighlightImage: '/img/games/highlightName/nameStarWars.png',
    altHighlightImage: 'Star Wars Highlight',
  },
];
