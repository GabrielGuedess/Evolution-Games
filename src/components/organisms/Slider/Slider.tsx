import {
  Parallax,
  Keyboard,
  Autoplay,
  EffectFade,
  Navigation,
  Pagination,
} from 'swiper';
import { Swiper, SwiperSlide } from 'swiper/react';

import MediaMatch from 'components/atoms/MediaMatch/MediaMatch';
import { ScrollDown } from 'components/atoms/ScrollDown/ScrollDown';
import { Slide, SlideProps } from 'components/molecules/Slide/Slide';

import * as S from './Slider.styles';

export type SliderProps = {
  slides: SlideProps[];
};

export const Slider = ({ slides }: SliderProps) => (
  <S.Wrapper>
    <Swiper
      pagination={{
        dynamicBullets: true,
        clickable: true,
      }}
      autoplay={{
        delay: 4000,
        waitForTransition: true,
        disableOnInteraction: false,
      }}
      keyboard={{
        enabled: true,
      }}
      parallax
      loop
      grabCursor
      effect="fade"
      modules={[
        Parallax,
        EffectFade,
        Autoplay,
        Navigation,
        Pagination,
        Keyboard,
      ]}
    >
      {slides.map(slide => (
        <SwiperSlide key={slide.altSlideImage}>
          <Slide {...slide} />
        </SwiperSlide>
      ))}
    </Swiper>

    <MediaMatch greaterThan="medium">
      <ScrollDown />
    </MediaMatch>
  </S.Wrapper>
);
