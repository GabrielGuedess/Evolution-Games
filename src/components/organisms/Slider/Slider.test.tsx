import { renderWithProviders } from 'utils/tests/helpers';

import { screen } from '@testing-library/react';

import sliderMock from './mock';
import { Slider } from './Slider';

jest.mock('swiper/css', jest.fn());
jest.mock('swiper/css/effect-fade', jest.fn());
jest.mock('swiper/css/pagination', jest.fn());

jest.mock('swiper', () => ({
  __esModule: true,
  Parallax: jest.fn(),
  Keyboard: jest.fn(),
  Autoplay: jest.fn(),
  EffectFade: jest.fn(),
  Navigation: jest.fn(),
  Pagination: jest.fn(),
}));

jest.mock('swiper/react', () => ({
  __esModule: true,
  Swiper: function Mock({ children }: { children: React.ReactNode }) {
    return (
      <div data-testid="Swiper">
        <div data-testid="Swiper-Slide">{children}</div>
      </div>
    );
  },
  SwiperSlide: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="Swiper-Slide">{children}</div>;
  },
}));

describe('<Slider />', () => {
  it('should render the Swiper correctly', () => {
    // Arrange
    const { container } = renderWithProviders(<Slider slides={sliderMock} />);

    // Assert
    expect(screen.getByTestId('Swiper')).toBeInTheDocument();

    expect(screen.getAllByTestId('Swiper-Slide')).toHaveLength(
      sliderMock.length + 1,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
