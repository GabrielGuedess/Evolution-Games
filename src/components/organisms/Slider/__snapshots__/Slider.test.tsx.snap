// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Slider /> should render the Swiper correctly 1'] = `
.c5 {
  display: none;
}

.c6 {
  position: absolute;
  -webkit-inset: auto auto 3rem 3rem;
  -ms-inset: auto auto 3rem 3rem;
  inset: auto auto 3rem 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  z-index: 10;
}

.c6 div {
  position: relative;
  width: 2rem;
  height: 4rem;
  border: 0.2rem solid #FFFFFF;
  border-radius: 5rem;
}

.c6 div:before {
  content: '';
  position: absolute;
  left: 50%;
  top: 0.6rem;
  width: 0.6rem;
  height: 0.6rem;
  margin-left: -0.3rem;
  background-color: #FFFFFF;
  border-radius: 100%;
  -webkit-animation: bPWYhV 2s infinite;
  animation: bPWYhV 2s infinite;
}

.c3 {
  max-width: 165.6rem;
  margin: 0 1.6rem;
}

.c1 {
  height: 100%;
  height: 100%;
}

.c2 {
  position: relative;
  height: 100%;
  height: 100%;
}

.c2 img {
  -webkit-mask-image: linear-gradient( 180deg, transparent 4%, #FFFFFF 56%, transparent 100% );
  mask-image: linear-gradient( 180deg, transparent 4%, #FFFFFF 56%, transparent 100% );
}

.c4 {
  position: absolute;
  inset: 0;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  padding-top: 12rem;
  z-index: 10;
}

.c0 {
  position: relative;
  width: 100%;
  height: 100vh;
}

.c0 .swiper {
  width: 100%;
  height: 100%;
}

.c0 .swiper-slide {
  background: #04040C;
}

.c0 .swiper-pagination {
  margin-bottom: 3rem;
}

.c0 .swiper-pagination span.swiper-pagination-bullet.swiper-pagination-bullet-active {
  background-color: #855aff;
  opacity: 1;
}

.c0 .swiper-pagination .swiper-pagination-bullet {
  background-color: #FFFFFF;
  opacity: 1;
}

@media (min-width:768px) {
  .c5 {
    display: block;
  }
}

@media (min-width:1170px) {
  .c3 {
    margin: 0 3.2rem;
  }
}

@media (min-width:1440px) {
  .c3 {
    margin: 0 4.8rem;
  }
}

@media (min-width:1656px) {
  .c3 {
    margin-left: auto;
    margin-right: auto;
  }
}

@media (min-width:768px) {
  .c4 {
    padding-top: 20rem;
  }
}

<section
  class="c0"
>
  <div
    data-testid="Swiper"
  >
    <div
      data-testid="Swiper-Slide"
    >
      <div
        data-testid="Swiper-Slide"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="The Last Of Us"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="c3 c4"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27768%27%20height=%27296%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="The Last Of Us Highlight"
                data-nimg="intrinsic"
                data-swiper-parallax="-300"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </div>
      <div
        data-testid="Swiper-Slide"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="God Of War"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="c3 c4"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27768%27%20height=%27296%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="God Of War Highlight"
                data-nimg="intrinsic"
                data-swiper-parallax="-300"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </div>
      <div
        data-testid="Swiper-Slide"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Resident Evil Village"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="c3 c4"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27768%27%20height=%27296%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Resident Evil Village Highlight"
                data-nimg="intrinsic"
                data-swiper-parallax="-300"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </div>
      <div
        data-testid="Swiper-Slide"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Star Wars"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
              />
              <noscript />
            </span>
          </div>
          <div
            class="c3 c4"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27768%27%20height=%27296%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="Star Wars Highlight"
                data-nimg="intrinsic"
                data-swiper-parallax="-300"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
              />
              <noscript />
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c5"
  >
    <button
      aria-label="Scroll Button"
      class="c6"
    >
      <div />
    </button>
  </div>
</section>
`;
